src/components/WeatherWidgetCurrentData.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <section class="weather-info">
+    <div class="weather-info__image-wrapper">
+      <img
+        class="weather-info__image"
+        :src="props.weatherIcon"
+        :alt="`${props.weatherDescription} Weather Image`"
+      />
+      <p class="weather-info__description">
+        {{ props.weatherDescription }}
+      </p>
+    </div>
+    <div class="weather-info__temperature-wrapper">
+      <h3 class="weather-info__temperature-info">
+        {{ props.temperature }}&deg;C
+      </h3>
+    </div>
+    <ul class="weather-info__additional-info">
+      <li
+        class="weather-info__additional-info-item"
+        v-for="(listItem, key) in sideListData"
+      >
+        {{ getListLabel(key) }}: {{ listItem }}
+      </li>
+    </ul>
+  </section>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+import { splitOnUppercaseChars } from "../utils.ts/splitOnUppercaseChars";
+
+export type WeatherData = {
+  temperature: number;
+  weatherIcon: string;
+  weatherDescription: string;
+  windSpeed: number;
+  pressure: number;
+  uvIndex: number;
+  date?: string;
+};
+
+const props = defineProps<WeatherData>();
+
+const sideListData = computed(() => {
+  const { weatherIcon, weatherDescription, temperature, date, ...listItems } =
```

Comment:
This is a bit verbose since we specify unrelated props here which might make it less readable. I would prefer to be more explicit here:

```
const { windSpeed, pressure, uvIndex } = props;

return { windSpeed, pressure, uvIndex };
```

Reply:
Resolved.